src/backups.py

```diff
@@ -531,7 +536,11 @@ def _on_s3_credential_changed(self, event: CredentialsChangedEvent):
 
     def _on_s3_credential_gone(self, _) -> None:
         if self.charm.unit.is_leader():
-            self.charm.app_peer_data.update({"stanza": "", "init-pgbackrest": ""})
+            self.charm.app_peer_data.update({
+                "stanza": "",
+                "init-pgbackrest": "",
+                "require-change-bucket-after-restore": "",
```

Comment:
I think it's better to not use this field as some identifier for bad S3 stanza, because this issue can be triggered on several stanzas simultaneously, and it will be hard to keep all this information here. Also, this potentially would cause issues for scenarios when one cluster tries to use another cluster's S3 stanza. So, creating universal solution to identify eligibility to use stanza (like by WAL parity as in the conversation above) is preferred.

Reply:
I will resolve this for now to keep all of this conversation in one place.